src/frontend/components/app/sort-link.tsx

```diff
@@ -13,33 +13,32 @@ export type SortLinkProps = {
   sortBy?: string;
 }
 
-class SortLink extends React.PureComponent<SortLinkProps & RouteComponentProps> {
-  constructor(props) {
-    super(props)
-    this.isActive = this.isActive.bind(this)
-  }
-
-  isActive(): boolean {
-    const { sortBy, property } = this.props
-    return sortBy === property.propertyPath
-  }
-
-  render(): ReactNode {
-    const { property, location, direction } = this.props
-    const query = new URLSearchParams(location.search)
-    const oppositeDirection = (this.isActive() && direction === 'asc') ? 'desc' : 'asc'
-    const sortedByIcon = `Caret${direction === 'asc' ? 'Up' : 'Down'}`
-
-    query.set('direction', oppositeDirection)
-    query.set('sortBy', property.propertyPath)
-
-    return (
-      <NavLink to={{ search: query.toString() }} className={cssClass('SortLink')}>
-        {property.label}
-        {this.isActive() ? (<Icon icon={sortedByIcon} color="primary100" ml="default" />) : ''}
-      </NavLink>
-    )
-  }
+const SortLink: React.FC<SortLinkProps> = (props) => {
+  const { sortBy, property, direction } = props
+  const location = useLocation()
+
+  const isActive = useMemo(() => sortBy === property.propertyPath, [sortBy, property])
+
+  const query = new URLSearchParams(location.search)
+  const oppositeDirection = (isActive && direction === 'asc') ? 'desc' : 'asc'
```

Comment:
Enum for `desc` and `asc` would be great

Reply:
This is currently used in too many places, I'll focus on rewriting this when working on AdminJS v7, otherwise Jarek and others will have more work to do when synchronizing changes while they split frontend to another repo